tools-api/src/ideogram/ideogram.service.ts

```diff
@@ -14,80 +15,127 @@ export class IdeogramService {
+      return option as MagicPromptOption;
+    }
+    throw new BadRequestException(`Invalid magic prompt option: ${option}`);
+  }
+
+  private validateAndConvertStyleType(style: string): StyleType {
+    if (Object.values(StyleType).includes(style as StyleType)) {
+      return style as StyleType;
+    }
+    throw new BadRequestException(`Invalid style type: ${style}`);
+  }
+
+  async generateIdeogram(generateDto: GenerateDTO, userId?: string) {
+    this.logger.log(`[START] Generating ideogram with prompt: "${generateDto.prompt.substring(0, 50)}..." for user: ${userId || 'anonymous'}`);
+    
+    try {
+      const response = await firstValueFrom(
+        this.httpService.post(
+          `${this.baseUrl}/generate`,
+          {
+            image_request: generateDto,
           },
-        }
-      )
-    );
+          {
+            headers: {
+              'Api-key': this.apiKey,
+              'Content-Type': 'application/json',
+            },
+          }
+        )
+      );
 
-    const imageData = response.data?.data?.[0];
+      this.logger.log(`[SUCCESS] Received response from Ideogram API for generation`);
+      const imageData = response.data?.data?.[0];
 
-    const { url } = await this.bucketService.uploadImageToBucket(imageData.url);
+      this.logger.log(`[PROCESSING] Uploading generated image to bucket`);
+      const { url } = await this.bucketService.uploadImageToBucket(imageData.url);
+      this.logger.log(`[SUCCESS] Image uploaded to bucket: ${url.substring(0, 50)}...`);
 
-    return {
-      ...imageData,
-      url,
-    };
+      // Store the creation if userId is provided
+      if (userId) {
+        this.logger.log(`[STORING] Saving original creation to database for user: ${userId}`);
+        try {
+          const creation = await this.creationService.createCreation(userId, {
+            imageUrl: url,
+            type: CreationType.ORIGINAL,
+            prompt: generateDto.prompt,
+            model: this.validateAndConvertModel(generateDto.model),
+            aspectRatio: this.validateAndConvertAspectRatio(generateDto.aspect_ratio),
+            magicPromptOption: this.validateAndConvertMagicPromptOption(generateDto.magic_prompt_option),
+            styleType: this.validateAndConvertStyleType(generateDto.style_type),
+            seed: generateDto.seed
+          });
+          this.logger.log(`[SUCCESS] Original creation stored successfully with ID: ${creation['_id']}`);
+        } catch (error) {
+          this.logger.error(`[ERROR] Failed to store original creation: ${error.message}`);
+          // Continue execution even if storage fails
+        }
+      } else {
+        this.logger.warn(`[SKIPPED] No userId provided, skipping creation storage`);
+      }
+
+      return {
+        ...imageData,
+        url,
+      };
+    } catch (error) {
+      this.logger.error(`[ERROR] Failed to generate ideogram: ${error.message}`);
+      throw error;
```

Comment:
better to be specific with the error, for example in this case would ServerErrorExecption